powershell/ql/lib/semmle/code/powershell/security/CommandInjectionCustomizations.qll

```diff
@@ -55,11 +61,169 @@ module CommandInjection {
         call.getAnArgument() = this
       )
     }
+    override string getSinkType() {
+      result = "call to Add-Type"
+    } 
   }
 
+  class InvokeScriptSink extends Sink {
+    InvokeScriptSink() { 
+        exists(API::Node call | 
+            API::getTopLevelMember("executioncontext").getMember("invokecommand").getMethod("invokescript") = call and
+            this = call.getArgument(_).asSink()
+        )
+    }
+    override string getSinkType(){
+        result = "call to InvokeScript"
+    } 
+}
+
+class CreateNestedPipelineSink extends Sink {
+    CreateNestedPipelineSink() { 
+        exists(API::Node call | 
+            API::getTopLevelMember("host").getMember("runspace").getMethod("createnestedpipeline") = call and
+            this = call.getArgument(_).asSink()
+        )
+    }
+    override string getSinkType(){
+        result = "call to CreateNestedPipeline"
+    }
+}
+
+class AddScriptInvokeSink extends Sink {
+    AddScriptInvokeSink() { 
+        exists(InvokeMemberExpr ie | 
+            this.asExpr().getExpr() = ie.getAnArgument() and
+            ie.getName() = "AddScript" and
+            ie.getQualifier().(InvokeMemberExpr).getName() = "Create" and
+            ie.getQualifier().getAChild().toString() = "PowerShell" and
```

Comment:
GitHub really dislikes using `toString` in query logic since the values returned by `toString` are often changed for performance / usability reasons. So we should probably follow their rules going forward